index.ts

```diff
@@ -1,3 +1,21 @@
+// Global error handler
+process.on('uncaughtException', (err) => {
+  console.error('Uncaught Exception:', err);
+  process.exit(1); // Optional: exit the process after logging the error
+});
+
+process.on('unhandledRejection', (reason, promise) => {
+  console.error('Unhandled Rejection at:', promise, 'reason:', reason);
+  process.exit(1); // Optional: exit the process after logging the error
+});
+
+// Graceful shutdown on SIGTERM
+process.on('SIGTERM', () => {
+  console.log('SIGTERM signal received: closing HTTP server');
+  // Perform any necessary cleanup here
+  process.exit(0);
```

Comment:
not sure what the exit() parameter means, so might be useful to add a comment about that, and also if 0/1 are representing a boolean type, I'd prefer true/false to be used, as it explicitly represents the type.